module/zfs/dsl_dataset.c

```diff
@@ -3231,6 +3235,8 @@ dsl_dataset_rollback_sync(void *arg, dmu_tx_t *tx)
 
 	VERIFY0(dsl_dataset_hold(dp, ddra->ddra_fsname, FTAG, &ds));
 
+	dsl_dir_cancel_waiters(ds->ds_dir);
```

Comment:
since we do this in dsl_dataset_clone_swap_sync_impl, why do we also need to do it here?